src/chain-data/context.tsx

```diff
@@ -35,4 +37,37 @@ const ChainDataContextProvider = (props: { children: ReactNode }) => {
 
 export default ChainDataContextProvider;
 
-export const useChainData = () => useContext(ChainDataContext);
+export const useChainData = () => {
+  const data = useContext(ChainDataContext);
+  const { chain } = useNetwork();
+  const provider = useProvider({ chainId: chain?.id });
```

Comment:
I believe you don't need to use `chainId`

Reply:
Removing them seems to work as expected 👍